lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -258,18 +442,29 @@ def _call(self, method: Callable, json_data: JSON):
             es_request = self.assemble(json_data, context)
             path = es_request.get("path")
             url = urljoin(self.es_url, path)
-            self.logger.debug("ASSEMBLE returned URL {}", url)
+            self.logger.info(
+                "ASSEMBLE returned URL {!r}, {!r}", url, es_request.get("json")
+            )
+        except NoSelectedDatasets as e:
+            # The query builder reports that no datasets can be found; skip
+            # the Elasticsearch query and tell postprocess to build an empty
+            # client response as appropriate.
+            self.logger.info("{} query builder reports no matches: {}", klasname, e)
+            context["NODATA"] = True
         except Exception as e:
             self.logger.exception("{} assembly failed: {}", klasname, e)
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
         try:
-            # perform the Elasticsearch query
-            es_response = method(url, **es_request["kwargs"])
-            self.logger.debug(
-                "ES query response {}:{}", es_response.reason, es_response.status_code
-            )
-            es_response.raise_for_status()
+            if not context.get("NODATA", False):
```

Comment:
My mood changes, but I often dislike relying on implicit false-y logic and prefer to make it explicit. Putting the `False` here was admittedly a momentary whim, but it wasn't entirely random.

Reply:
> I often dislike relying on implicit false-y logic and prefer to make it explicit.

I respect and support that.  But, if that's the right approach here, then it seems to me that it would be the right approach in all the other locations which followed the other whim.  :wink: